### PR TITLE
NMS-8438: Upgrade PostgreSQL Driver and C3P0

### PR DESCRIPTION
**Pull request by agalue**
Upgrade PostgreSQL Driver and C3P0

* JIRA: http://issues.opennms.org/browse/NMS-8438
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS748

### CODE REVIEW

File: container/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -98,11 +98,11 @@
       <bundle>wrap:mvn:org.apache.xmlgraphics/batik-xml/1.7</bundle>
     </feature>
 
-    <feature name="c3p0" description="c3p0" version="0.9.1.2">
+    <feature name="c3p0" description="c3p0" version="${c3p0Version}">
       <feature>postgresql</feature>
 
       <!-- Add all possible JDBC drivers to the c3p0 import list so that it can classload them -->
-      <bundle>wrap:mvn:c3p0/c3p0/0.9.1.2$Bundle-SymbolicName=c3p0&amp;Bundle-Version=0.9.1.2&amp;Import-Package=javax.management,javax.naming,javax.naming.spi,javax.sql,javax.xml.parsers,org.apache.log4j,org.postgresql,org.w3c.dom</bundle>
+      <bundle>wrap:mvn:com.mchange/c3p0/${c3p0Version}$Bundle-SymbolicName=c3p0&amp;Bundle-Version=${c3p0Version}&amp;Import-Package=javax.management,javax.naming,javax.naming.spi,javax.sql,javax.xml.parsers,org.apache.log4j,org.postgresql,org.w3c.dom</bundle>
```
Comment (soleger) [inline container/features/src/main/resources/features.xml]: I think that you can update the Import-Package section here:

* Remove javax.naming.spi
* Remove org.apache.log4j

Can you make these changes and make sure that the Karaf integration tests work before merging this? Go into opennms-full-assembly and get rid of the @Ignore annotations temporarily and then run the tests as usual (mvn -DskipITs=false clean install). Thanks!

Comment (agalue) [inline container/features/src/main/resources/features.xml]: Sure thing, but you have to guide me. That feature was there, I haven't added it. Also, it is not loaded at runtime (after SSH to port 8101 at runtime). Anything related with Karaf/OSGi is new to me, so I need a step by step procedure if you want me to do something ;)

Comment (agalue) [inline container/features/src/main/resources/features.xml]: I think I got about what you said about removing the things from the Import-Package, but there is nothing inside opennms-full-assembly related with the word Ignore (this is based on foundation-2016 not develop). I did "git grep Ignore" inside that folder, and returned nothing. What should I do there ?

Comment (agalue) [inline container/features/src/main/resources/features.xml]: @soleger do you want me to temporary apply this patch to develop and execute the Karaf Integration tests, to be sure they pass ?

Comment (soleger) [inline container/features/src/main/resources/features.xml]: Yes, exactly. The Karaf integation tests are only in develop, sorry for the confusion. The c3p0 feature is not loaded on the OpenNMS side but it is loaded to satisfy dependencies inside Minion so we still need to test it to make sure it doesn't blow up. :)

Comment (agalue) [inline container/features/src/main/resources/features.xml]: > On May 25, 2016, at 9:29 AM, Seth Leger <notifications@github.com> wrote:
> 
> Yes, exactly. The Karaf integation tests are only in develop, sorry for the confusion.

No worries.

> The c3p0 feature is not loaded on the OpenNMS side but it is loaded to satisfy dependencies inside Minion so we still need to test it to make sure it doesn't blow up. :)


Cool! I'm going to temporary apply the changes locally on develop to execute the Karaf tests.

**Comment by agalue** [comment]
It says:
Tests run: 73, Failures: 0, Errors: 8, Skipped: 0

I'll check which of them has errors, but I'm wondering: without the changes, are all those tests passing on develop ? It took almost 90 minutes to run all those tests.

**Comment by agalue** [comment]
Hmmm,

Could not start bundle wrap:mvn:org.postgresql/postgresql/9.4.1208 in feature(s) postgresql-9.4.1208: Unresolved constraint in bundle org.postgresql.jdbc42 [97]: Unable to resolve 97.0: missing requirement [97.0] osgi.wiring.package; (osgi.wiring.package=javax.transaction.xa)

Any clues ?

**Comment by agalue** [comment]
@soleger It is better now, after your change ...
Tests run: 73, Failures: 0, Errors: 4, Skipped: 0

Checking the surfire-report details, I can see errors but none of them related with PostgreSQL/C3P0, so I think we're OK. Let me know if you want to check them.

**Pull request merged**